packages/core/src/components/forms/numericInput.tsx

```diff
@@ -256,16 +271,19 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
     }
 
     public componentDidUpdate() {
-        if (this.state.shouldSelectAfterUpdate) {
+        if (this.shouldSelectAfterUpdate) {
             this.inputElement.setSelectionRange(0, this.state.value.length);
         }
     }
 
     protected validateProps(nextProps: HTMLInputProps & INumericInputProps) {
-        const { majorStepSize, max, min, minorStepSize, stepSize } = nextProps;
-        if (min && max && min >= max) {
+        const { clampValueOnBlur, majorStepSize, max, min, minorStepSize, stepSize } = nextProps;
+        if (min != null && max != null && min >= max) {
             throw new Error(Errors.NUMERIC_INPUT_MIN_MAX);
         }
+        if (clampValueOnBlur && min == null && max == null) {
+            throw new Error(Errors.NUMERIC_INPUT_CLAMP_VALUE_ON_BLUR);
```

Comment:
is this actually a failure mode? or more of a warning?

seems like it will not explode: https://github.com/palantir/blueprint/pull/1057/files#diff-e7ce5ec58792f4f496c6767dc39ae655R510

is this check even necessary?